microcosm_pubsub/codecs.py

```diff
@@ -39,39 +40,17 @@ class PubSubMessageSchema(Schema):
     Base schema for messages, including a media type.
 
     """
-    media_type = fields.Method(
-        # called by dump
-        serialize="serialize_media_type",
-        # called by load
-        deserialize="deserialize_media_type",
-        attribute="mediaType",
-        # need to set missing to non-None or marshmallow won't call the deserialize function
-        missing=DEFAULT_MEDIA_TYPE,
+    MEDIA_TYPE = DEFAULT_MEDIA_TYPE
+
+    mediaType = fields.String(
+        attribute="media_type",
+        required=False,
     )
-    opaque_data = fields.Dict(
-        attribute="opaqueData",
+    opaqueData = fields.Dict(
+        attribute="opaque_data",
         required=False,
     )
 
-    def serialize_media_type(self, message):
-        """
-        Fetch the media type from the message.
-
-        """
-        try:
-            return message["mediaType"]
-        except KeyError:
-            raise ValidationError("Message did not define a media type")
```

Comment:
There was an odd level indecision here regarding whether or not all published messages required a media_type, and whether it should be present at the time of message encoding. I'm taking the stance here that all messages require the media type to be explicitly-specified prior to creation, and treated as such (i.e no implicit loading of media types).